From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Miles Chen <miles.chen@mediatek.com>
Date: Wed, 27 Jan 2021 12:31:23 +0800
Subject: ANDROID: GKI: select DMADEVICES if GKI_HIDDEN_SND_SOC_CONFIGS is set

DMA_VIRTUAL_CHANNELS depends on DMADEVICES and DMA_VIRTUAL_CHANNELS
is selected by GKI_HIDDEN_SND_SOC_CONFIGS.

To fix unmet dependency in x86_64, select DMADEVICES under
GKI_HIDDEN_SND_SOC_CONFIGS.

Also, drop CONFIG_DMADEVICES=y in arch/arm64/configs/gki_defconfig.

WARNING: unmet direct dependencies detected for DMA_VIRTUAL_CHANNELS
  Depends on [n]: DMADEVICES [=n]
    Selected by [y]:
      - GKI_HIDDEN_SND_SOC_CONFIGS [=y]

Bug: 174628645
Signed-off-by: Miles Chen <miles.chen@mediatek.com>
Change-Id: I1de491c10542f1b7c9c8754c65415b028d3bc922
---
 arch/arm64/configs/gki_defconfig | 1 -
 init/Kconfig.gki                 | 1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -449,7 +449,6 @@ CONFIG_EDAC=y
 CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_PL030=y
 CONFIG_RTC_DRV_PL031=y
-CONFIG_DMADEVICES=y
 CONFIG_DMABUF_HEAPS=y
 CONFIG_DMABUF_HEAPS_SYSTEM=y
 CONFIG_UIO=y
diff --git a/init/Kconfig.gki b/init/Kconfig.gki
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -48,6 +48,7 @@ config GKI_HIDDEN_SND_SOC_CONFIGS
 	select SND_SOC_GENERIC_DMAENGINE_PCM if (SND_SOC && SND)
 	select SND_PCM_IEC958
 	select SND_SOC_COMPRESS if (SND_SOC && SND)
+	select DMADEVICES
 	select DMA_VIRTUAL_CHANNELS
 	help
 	  Dummy config option used to enable hidden SND_SOC configs.
